### ISSUE TITLE
employee view only for cashups that have been opened

### ISSUE THREAD

**Issue by andriux1990** [opensourcepos/opensourcepos#2775]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [] Reporting an issue of an unmodified OSPOS installation
- [] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 
- OSPOS git commit hash is: 
- PHP version is: 
- MySQL or MariaDB version is: 
- OS and version is: 
- WebServer is: 
- Selected language is: 
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is: 

### Issue / Bug  / Question / New Feature

Please write your issue here. If a bug, please make sure to provide as much information as possible including configuration settings (e.g. Decimals set, Tax mode), language and steps to reproduce the bug.
-----------------------------------------------------------------------------------------------------------------------------------
I have assigned two computers to two employees connected to ospos under the same db.
I use cashup but when two users are connected to the same cashup system it does not work ...because I mix the two boxes (cashups).

It would be super that in cashup when a new cashup is opened this cashup I cannot change or modify another employee from its user. And if there were 10 employees and 10 cashups were opened, each employee would see only the one that had been opened.

**Comment by odiea** [comment]
You forget that OSPOS is really a Single User System.

**Comment by andriux1990** [comment]
> You forget that OSPOS is really a Single User System.

I know, I am very aware of that, but I have loved ospos so much that I was looking for a way to do it.

**Comment by jekkos** [comment]
You could solve this by linking the cashup to a location. I wouldn't link it to an employee as that does not make sense. In fact this is something you need a multi-tenant setup for indeed.

**Comment by andriux1990** [comment]
@jekkos you're right. the best way would be with location, and in expenses it could also be done with location. so that the cashups takes the expenses that I have made only this user and not the expenses that another user has made.

and the user with all the locations can see all the cashups and expenses

**Comment by odiea** [comment]
The issue I see with using locations is what if the location has multiple Registers ? Each employee would need to do their own cashup.

**Comment by andriux1990** [comment]
What exactly do you mean by this? In fact, the employee can see the sales of each employee in the reports. And you can also see the total sales total where all records of all employees are located.

The problem with cash and expenses is that any employee can change the information of another employee. Suppose there is an employee with bad intentions, you can do wrong with that and make changes to the other employee, or do it on purpose.

correcting these things would be ideal and better to use cashups and expenses with the locations such as sales and purchases can be done.

**Comment by odiea** [comment]
What I tried to say was Yes filtering it by location would work if the Location only has 1 register. When the location has more then 1 Register I think it should be filtered by Employee also.

**Comment by odiea** [comment]
In employees I now have a checkbox for admin,manager and clerk to show what position they hold. 
 I have each employee isolated to just show their sales, cashups and expenses.  What I would like to do is use the cashups open employee id to be used as the current employee id instead of using $employee_id = $this->session->userdata('person_id');.  Anyone have an idea on how to extract that info from Cashups?  The reason is when the manager or admin views the Cashups his employee id is being used . And that shows all Sales and Expenses.
```
if($position == 2)					
		{
				$employee_id = $this->session->userdata('person_id');//keeper
				$this->db->where('sales.employee_id',$employee_id);//keeper
		}
```

**Comment by odiea** [comment]
I will let you know once I figure out the admin or manager cashup issue. The cashups look ok once they are closed but not before if the manager or admin want to check to see how an employee is doing.

**Comment by andriux1990** [comment]
@odiea. any news, did you achieve any progress?

**Comment by odiea** [comment]
No. Still not showing correct amounts until the Cashup is closed.

**Comment by mckaygerhard** [comment]
i reviewed that @andriux1990  :

> I have assigned two computers to two employees connected to ospos under the same db.
> I use cashup but when two users are connected to the same cashup system it does not work ...because I mix the two boxes (cashups).

it looks like a multitier case! so if we can sole that! ospos can be used as a multilocation with only one database .. as @jekkos  said: 

> You could solve this by linking the cashup to a location.

that a hint for multiuser management! and when ospos team allow the cashup !" the multiuser case comes with that feature @odiea  so .. we have now a big problem in the future!

> 
> It would be super that in cashup when a new cashup is opened this cashup I cannot change or modify another employee from its user. And if there were 10 employees and 10 cashups were opened, each employee would see only the one that had been opened.

**Comment by andriux1990** [comment]
of course yes ... this would be great .. and I think ospos is not far from this.

@odiea Do you have any progress

**Comment by mckaygerhard** [comment]
the problems @andriux1990  **are that ospos developers seems are bit of confused: lest check history:**

1. in past 2015 there's a first attemp to make multi tier installations at this issue #150 that seems was the first intent to make ospos multi-install capable..
2. amonth the history multiple times was asked in some ways as in #1670 #1113 and #1664  but the most interesting response are in the comment of #1937 "i will not work for free" of @daN4cat  the "lead architect" adn i understand their reasons.. he wants money.. 
3. as a note in the first attemp issue was proposet the sync amont databases.. a mess of course cos are petty easy beetween two databases.. but 3, 4? a crap of course.. specially with the poor engine named mysql  :smile:  :smile:
3. later the most recent was issue #2226 where that behaviour was proposed without sync replication,, but with multiple instances shared.. this last time was not rejected.. and the question was not answered, but in fact it's "no" : ospos cannot setup with multiple instances.. also there's no accounting support inside (a very important note about)
 
HEre we have two problems but the main one are the database choosen: mysql. a poor engine for multi-sync purposes.. it works well in master-slave but multi-master nah! here sybase are the king of replication.. 

so my popos will be commented in the issue #150 .. please go here @andriux1990 if i cannot comment (yes you knowed!) i'll open a new issue

**Comment by andriux1990** [comment]
what do you think can be the solution?

**Comment by mckaygerhard** [comment]
solution need of the recent comments i posted in the #150 ... checked and later then we can made easyle the cashup per user and per location
